usr.bin/sockstat/sockstat.c

```diff
@@ -1107,235 +1093,420 @@ sctp_path_state(int state)
 	}
 }
 
+static int
+format_unix_faddr(struct addr *faddr, char *buf, size_t bufsize) {
+	size_t pos = 0;
+	/* Remote peer we connect(2) to, if any. */
+	if (faddr->conn != 0) {
+		struct sock *p;
+		strlcpy(buf, "-> ", bufsize);
+		pos += 3;
+		p = RB_FIND(pcbs_t, &pcbs,
+			&(struct sock){ .pcb = faddr->conn });
+		if (__predict_false(p == NULL)) {
+			/* XXGL: can this happen at all? */
+			pos += snprintf(buf == NULL ? NULL : buf+pos,
+				buf == NULL ? 0 : bufsize-pos, "??");
+		} else if (p->laddr->address.ss_len == 0) {
+			struct file *f;
+			f = RB_FIND(files_t, &ftree,
+				&(struct file){ .xf_data =
+				p->socket });
+			if (f != NULL) {
+				pos += snprintf(buf == NULL ? NULL : buf+pos,
+					buf == NULL ? 0 : bufsize-pos,
+					"[%lu %d]", (u_long)f->xf_pid, f->xf_fd);
```

Comment:
Instead of trying to keep track of `pos`, you could print lines like this into a stack buffer, then use `strlcat` to copy that to `buf`.  Which technique you prefer is your choice.